File: Domain/Order.cs
using Domain.Enums;

namespace Domain
{
    public class Order
    {
        public int Id { get; set; }
        public int CustomerDetailsId { get; set; }
        public DateTime OrderDate { get; set; }
        public OrderStatus Status { get; set; }
        public int PaymentMethodId { get; set; } 
        public int ShippingMethodId { get; set; } 

        public CustomerDetails CustomerDetails { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        public ShippingMethod ShippingMethod { get; set; }
        public ICollection<OrderItem> Items { get; set; }
    }
}

File: client-zespolowy/src/app/models/onlineshop/Order.ts
import { OrderStatus } from "../enums/OrderStatus";
import { AccountDetails } from "./AccountDetails";
import { CartItem } from "./Cart";
import { OrderItem } from "./OrderItem";
import { PaymentMethod } from "./PaymentMethod";
import { ShippingMethod } from "./ShippingMethod";

export interface Order {
    id: number;
    orderDate: Date;
    status: OrderStatus;
    paymentMethod: PaymentMethod;
    shippingMethod: ShippingMethod;
    userDetails: AccountDetails;
    items: OrderItem[];
}

export interface CreateOrder {
    paymentMethodId: number,
    shippingMethodId: number,
    items: CartItem[],
}